pkg/cmd/admin/migrate/images/images.go

```diff
@@ -0,0 +1,484 @@
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	internalMigrateImagesLong = `
+Migrate references to Docker images
+
+This command updates embedded Docker image references on the server in place. By default it
+will update image streams and images, and may be used to update resources with a pod template
+(deployments, replication controllers, daemon sets).
+
+References are changed by providing a mapping between a source registry and name and the
+desired registry and name. Either name or registry can be set to '*' to change all values.
+The registry value "docker.io" is special and will handle any image reference that refers to
+the DockerHub. You may pass multiple mappings - the first matching mapping will be applied
+per resource.
+
+The following resource types may be migrated by this command:
+
+* images               * daemonsets
+* imagestreams         * jobs
+* buildconfigs         * replicationcontrollers
+* deploymentconfigs    * pods
+* secrets (docker)
+
+Only images, imagestreams, and secrets are updated by default. Updating images and image
+streams requires administrative privileges.`
+
+	internalMigrateImagesExample = `  # Perform a dry-run of migrating all "docker.io" references to "myregistry.com"
+  %[1]s docker.io/*=myregistry.com/*
+
+  # To actually perform the migration, the confirm flag must be appended
+  %[1]s docker.io/*=myregistry.com/* --confirm
+
+  # To see more details of what will be migrated, use the loglevel and output flags
+  %[1]s docker.io/*=myregistry.com/* --loglevel=2 -o yaml
+
+  # Migrate from a service IP to an internal service DNS name
+  %[1]s 172.30.1.54/*=registry.openshift.svc.cluster.local/*
+
+  # Migrate from a service IP to an internal service DNS name for all deployment configs and builds
+  %[1]s 172.30.1.54/*=registry.openshift.svc.cluster.local/* --include=buildconfigs,deploymentconfigs`
+)
+
+type MigrateImageReferenceOptions struct {
+	migrate.ResourceOptions
+
+	Client   client.Interface
+	Mappings ImageReferenceMappings
+}
+
+// NewCmdMigrateImageReferences implements a MigrateImages command
+func NewCmdMigrateImageReferences(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateImageReferenceOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:      in,
+			Out:     out,
+			ErrOut:  errout,
+			Include: []string{"imagestream", "image", "secrets"},
+		},
+	}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s REGISTRY/NAME=REGISTRY/NAME [...]", name),
+		Short:   "Update embedded Docker image references",
+		Long:    internalMigrateImagesLong,
+		Example: fmt.Sprintf(internalMigrateImagesExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			if err := options.Run(); err != nil {
+				// TODO: move met to kcmdutil
```

Comment:
Yeah, the rebase got my upstream fix.

On Thu, Jun 16, 2016 at 5:14 AM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/cmd/admin/migrate/images/images.go
> <https://github.com/openshift/origin/pull/9240#discussion_r67312076>:
>
> > +			In:      in,
> > +			Out:     out,
> > +			ErrOut:  errout,
> > +			Include: []string{"imagestream", "image", "secrets"},
> > +		},
> > +	}
> > +	cmd := &cobra.Command{
> > +		Use:     fmt.Sprintf("%s REGISTRY/NAME=REGISTRY/NAME [...]", name),
> > +		Short:   "Update embedded Docker image references",
> > +		Long:    internalMigrateImagesLong,
> > +		Example: fmt.Sprintf(internalMigrateImagesExample, fullName),
> > +		Run: func(cmd *cobra.Command, args []string) {
> > +			kcmdutil.CheckErr(options.Complete(f, cmd, args))
> > +			kcmdutil.CheckErr(options.Validate())
> > +			if err := options.Run(); err != nil {
> > +				// TODO: move met to kcmdutil
>
> I don't think it's needed. With the current implementation
> kcmutil.CheckErr should os.Exit(1) with that error, printing nothing
> (from looking into StandardErrorMessage from kcmdutil package).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9240/files/eb1eb8a1f37fe62714438a3a805545a98b82daec..053e3598622ec47ce57195dacc03aedc08e4489f#r67312076>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3zxxuIxre02vI8IzwzfJNmdGNccks5qMRPkgaJpZM4IxnZA>
> .
>